File: node.js/백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
'use strict';
const fs = require('fs');

let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
let [n, m] = input[0].split(' ').map(Number);


let baskets = Array.from({ length: n }, (_, i) => i + 1);


for (let k = 1; k <= m; k++) {
  let [i, j] = input[k].split(' ').map(Number);

  while (i < j) {
    let temp = baskets[i - 1];
    baskets[i - 1] = baskets[j - 1];
    baskets[j - 1] = temp;
    i++;
    j--;
  }
}


console.log(baskets.join(' '));

File: C99/백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);

    int baskets[N];
    for (int i = 0; i < N; i++) {
        baskets[i] = i + 1;
    }

    for (int m = 0; m < M; m++) {
        int i, j;
        scanf("%d %d", &i, &j);
        i--; 
        j--;

        while (i < j) {
            int temp = baskets[i];
            baskets[i] = baskets[j];
            baskets[j] = temp;
            i++;
            j--;
        }
    }

    for (int i = 0; i < N; i++) {
        printf("%d ", baskets[i]);
    }
    printf("\n");

    return 0;
}